envoy: Default to daemon set deployment from 1.16 (#30034)
cilium-team/cilium: Documentation/operations/upgrade.rst

1.16 Upgrade Notes
------------------

> * Cilium Envoy DaemonSet is now enabled by default, and existing in-container installs will be changed to DaemonSet mode
>   unless specifically opted out of. This can be done by disabling it manually manually by setting ``envoy.enabled=false``
>   accordingly.

Reviewer: ```suggestion
* Cilium Envoy DaemonSet is now enabled by default, and existing in-container installs
  will be changed to DaemonSet mode unless specifically opted out of. This can be done by
  disabling it manually by setting ``envoy.enabled=false`` accordingly.
```

Just to tidy up the formatting and remove the double "manually".

1.16 Upgrade Notes
------------------

> * Cilium Envoy daemon set is now enabled by default. If you are upgrading from a

Reviewer 2: Have you experimented with the `upgradeCompatibility` Helm value? In other words, if users set `upgradeCompatitibilty <= 1.15" (which tells us they are upgrading from an existing Cilium version), is there a way to disable the DaemonSet?

This is typically how we deal with breaking changes in Helm values. Most of them only apply to ConfigMaps, but depending on the complexity it might also be possible to use here:
https://github.com/cilium/cilium/blob/7318ce2d0d89a91227e3f313adebce892f3c388e/install/kubernetes/cilium/templates/cilium-configmap.yaml#L2-L85

Author: Yes, I looked at upgradeCompatibility flag before, but it seems to work for non-default attribute only, for example,   kubeProxyReplacement is commented in values.yaml.

Reviewer 2: Ah, fair enough!

> * Cilium Envoy daemon set is now enabled by default. If you want to preserve existing

Reviewer: Is this the case for all upgrades, or only for new installs?

Author: It will be for all upgrade pre 1.16.

Reviewer: I think it would be good to have something like this:
```suggestion
* Cilium Envoy DaemonSet is now enabled by default, and existing in-container installs will be changed to DaemonSet mode unless specifically opted out of. This can be done by disabling it manually...
```

Just a stronger callout so that people who do read the changelog understand the change.

Author: ah thanks, let me make one update for the above.

>   previous version, you need to enable it manually by setting ``envoy.enabled=false``

Reviewer 2: Shouldn't this be 
```suggestion
  previous version, you need to disable it manually by setting ``envoy.enabled=false``
```

And maybe mention "if you want to preserve existing behavior"?

Author: ah right, thanks a lot, let me push this change 👍
